example_configs/spark-3-0.yml

```diff
@@ -0,0 +1,131 @@
+rules:
+
+  # These come from the master
+  # Example: master.aliveWorkers
+  - pattern: "metrics<name=master\\.(.*), type=counters><>Value"
+    name: spark_master_$1
+
+  # These come from the worker
+  # Example: worker.coresFree
+  - pattern: "metrics<name=worker\\.(.*), type=counters><>Value"
+    name: spark_worker_$1
+
+  # These come from the application driver
+  # Example: app-20160809000059-0000.driver.DAGScheduler.stage.failedStages
+  - pattern: "metrics<name=(.*)\\.driver\\.(DAGScheduler|BlockManager|jvm)\\.(.*), type=gauges><>Value"
+    name: spark_driver_$2_$3
+    type: GAUGE
+    labels:
+      app_id: "$1"
+
+  # These come from the application driver
+  # Emulate timers for DAGScheduler like messagePRocessingTime
+  - pattern: "metrics<name=(.*)\\.driver\\.DAGScheduler\\.(.*), type=counters><>Count"
+    name: spark_driver_DAGScheduler_$2_count
+    type: COUNTER
+    labels:
+      app_id: "$1"
+
+  # HiveExternalCatalog is of type counter
```

Comment:
This is redundant.

Reply:
Could you be more specific? This is a spark driver metric, i didn't change there anything except for adding the metric type

Reply:
The comment isn't needed.